pkg/authorization/authorizer/authorizer.go

```diff
@@ -0,0 +1,516 @@
+package authorizer
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	klabels "github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	authenticationapi "github.com/openshift/origin/pkg/auth/api"
+	authcontext "github.com/openshift/origin/pkg/auth/context"
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	policyregistry "github.com/openshift/origin/pkg/authorization/registry/policy"
+	policybindingregistry "github.com/openshift/origin/pkg/authorization/registry/policybinding"
+)
+
+type openshiftAuthorizer struct {
+	masterAuthorizationNamespace string
+	policyRegistry               policyregistry.Registry
+	policyBindingRegistry        policybindingregistry.Registry
+}
+type openshiftAuthorizationAttributes struct {
+	user              authenticationapi.UserInfo
+	verb              string
+	resourceKind      string
+	namespace         string
```

Comment:
> Maybe I'm mixing conversations in my mind, and that was something we talked about later when discussing all possible inputs to an authorizer

Yes. That was for possible inputs to an authorizer.  Given that it is a relatively advanced case, I'd propose leaving that until we get to the fine grained aspects of policy.